docs/3to4.md

```diff
@@ -1,170 +1,179 @@
-  `null` raises conflict.
+- In the [`xadd`](https://redis.io/commands/XADD) method with StreamEntryID 
+parameter, sending untyped `null` raises an exception.
 
   Casting the `null` to StreamEntryID (`(StreamEntryID) null`) resolves this issue.
 
-- In [`xrange`](https://redis.io/commands/XRANGE) and
+- In the [`xrange`](https://redis.io/commands/XRANGE) and
   [`xrevrange`](https://redis.io/commands/xrevrange) methods with StreamEntryID parameters, sending
-  untyped `null`s for both start and end parameters raises conflicts.
+  untyped `null`s for both start and end parameters raises an exception.
 
   Casting the `null`s to StreamEntryID (`(StreamEntryID) null`) resolves this issue.
 
-- The return type of Jedis.shutdown() is now `void`. Previously, ideally, it would return null.
-
-- `eval` and `evalsha` methods are now non-blocking. These have been blocking in Jedis 3.x.
-
-- HostAndPort.`localhost` variable has been removed.
-
-- These methods have been removed from HostAndPort class:
-  - extractParts
-  - parseString
-  - convertHost
-  - setLocalhost
-  - getLocalhost
-  - getLocalHostQuietly
-
-- These classes have been moved to the `redis.clients.jedis.args` package.
-  - BitOP
-  - GeoUnit
-  - ListPosition
-
-- These classes have been moved to the `redis.clients.jedis.params` package.
-  - BitPosParams
-  - ScanParams
-  - SortingParams
-  - ZParams
-
-- These classes have been moved to the `redis.clients.jedis.resps` package.
-  - AccessControlLogEntry
-  - AccessControlUser
-  - GeoRadiusResponse
-  - ScanResult
-  - Slowlog
-  - StreamConsumersInfo
-  - StreamEntry
-  - StreamGroupInfo
-  - StreamInfo
-  - StreamPendingEntry
-  - StreamPendingSummary
-  - Tuple
-
-- Jedis and JedisPool constructors with a String parameter, and no int parameter only support a URL
-  or URI string.
+- The return type of `Jedis.shutdown()` is now `void`. Previously, it would return null.
+
+- The `eval` and `evalsha` methods are now non-blocking. These methods were 
+blocking in Jedis 3.x.
+
+- The `HostAndPort.localhost` constant has been removed.
+
+- The following methods have been removed from HostAndPort class:
+  - `extractParts()`
+  - `parseString()`
+  - `convertHost()`
+  - `setLocalhost()`
+  - `getLocalhost()`
+  - `getLocalHostQuietly()`
+
+- The following classes have been moved to the `redis.clients.jedis.args` package.
+  - `BitOP`
+  - `GeoUnit`
+  - `ListPosition`
+
+- The following classes have been moved to the `redis.clients.jedis.params` package.
+  - `BitPosParams`
+  - `ScanParams`
+  - `SortingParams`
+  - `ZParams`
+
+- The following classes have been moved to the `redis.clients.jedis.resps` package.
+  - `AccessControlLogEntry`
+  - `AccessControlUser`
+  - `GeoRadiusResponse`
+  - `ScanResult`
+  - `Slowlog`
+  - `StreamConsumersInfo`
+  - `StreamEntry`
+  - `StreamGroupInfo`
+  - `StreamInfo`
+  - `StreamPendingEntry`
+  - `StreamPendingSummary`
+  - `Tuple`
```

Comment:
To other reviewers:

Too many code formatting annoys me. Am I the only one here?

PS: @banker This is not a change request.